code/datums/components/storage/concrete/bag_of_holding.dm

```diff
@@ -1,16 +1,16 @@
 /datum/component/storage/concrete/bluespace/bag_of_holding/handle_item_insertion(obj/item/W, prevent_warning = FALSE, mob/living/user)
 	var/atom/A = parent
 	if((istype(W, /obj/item/storage/backpack/holding) || count_by_type(W.GetAllContents(), /obj/item/storage/backpack/holding)))
+		var/safety = alert(user, "Doing this will have extremely dire consequences for the station and its crew. Be sure you know what you're doing.", "Put in [A.name]?", "Abort", "Proceed")
+		if(safety != "Proceed" || QDELETED(A) || QDELETED(W) || QDELETED(user) || !in_range(A, user) || user.incapacitated())

```

Comment:
Should switch the last two to `!user.canUseTopic(A, BE_CLOSE, NO_DEXTERY)` since this currently doesn't check if they're restrained.